enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,773 @@
+  affinity, any node selector term having a `matchExpression` element using the
+  key `kubernetes.io/arch` is preserved. For all the other node
+  selector terms, we append the architecture-specific `matchExpression`.
+- *Image inspection failure*: in the node affinity patching workflow, if there
+  are any errors during the image inspection process (authentication issues,
+  network errors, ...), the scheduling gate is removed, and no changes are made
+  to the pod. In this case, the scheduling will only be architecture-aware if
+  the user submitting the pod defines the affinity predicates explicitly.
+
+### API Extensions
+
+#### Mutating Admission Webhook and Scheduling Gate 
+The multiarch-manager-operator reconciles not-yet-scheduled pods using the
+scheduling readiness[^1] and pod mutable scheduling directives[^2] features.
+Since Kubernetes 1.27[^3] promoted those features to beta, no additional changes
+to the Openshift Cluster Operators are required to leverage them in the context
+of automatically setting the architecture-related node affinity of pods. As
+described in the Workflow Description section, the
+`SchedulingGateMutatingAdmissionWebhook` changes a pod at admission time. It
+adds the `multi-arch.openshift.io/arch-aware-placement` scheduling gate to newly
+created pods. The scheduling gate can only be added to pods at creation time.
+The `PodPlacementController` is responsible for deleting the scheduling gate
+from the pod after inspecting the architecture of the pod containers' images and
+patching the node affinity set in the pod. The controller inspecting the image
+has access to any secrets in the cluster and builds the credentials to access
+the registries consuming the inspected pod's `imagePullSecrets` field. The
+controller will also read the image content source policy objects once the
+support for restricted network scenarios are implemented. 
+
+#### Required RBAC permissions
+
+The service account running the operator components will need the following RBAC
+permissions:
+
+- Deployment (namespace-scoped)
+  - Watch, List, Get, Create, Update, Delete
+- Pod (cluster-scoped)
+  - Watch, List, Get, Update
+- Secrets (cluster-scoped)
+  - Watch, List, Get
+- ConfigMaps (namespace-scoped)
+  - Watch, List, Get
+- PodPlacementConfig CRD
+  - Watch, List, Get, Update
+- Imagestreams/layers (required for internal registry access)
+  - Get
+- operator.openshift.io/ImageContentSourcePolicy
+- config.openshift.io/ImageDigestMirrorSet
+- config.openshift.io/ImageTagMirrorSet
+  - Watch, List, Get
+- config.openshift.io/Image
+  - Watch, List, Get
+
+#### PodPlacementConfig CR
+The configuration CR of the operator allows the admins to set the following:
+- a list of `ExcludedNamespaces` listing the namespaces to exclude when running
+  the controller logic. Openshift and kube namespaces are excluded implicitly
+- a `skipTLSVerification` flag to completely disable the verification of the TLS
```

Comment:
I don’t think this should exist at all. Users should not be asked to manage TLS enforcement for the same registry in two places.

https://github.com/openshift/api/blob/b2e6ab3d6c69e3efb4e7d52184b5794b8359165b/config/v1/types_image.go#L115 already exists, so I would expect there to be _some_ way for this software to consume that value.

It doesn’t make much sense to me to introduce a new CR with a new field and to immediately plan to deprecate it. Just read the right data from the start.